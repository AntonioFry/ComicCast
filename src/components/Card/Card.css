.character-card {
  align-items: flex-end;
  background-size: 215px;
  background-repeat: no-repeat;
  border: 1px solid black;
  display: flex;
  height: 250px;
  
  width: 190px;
}

.character-card-img {
  position: relative;
  width: 170px;
  height: 250px;
  z-index: -1;
}

.character-card-text {
  color: white;
  text-decoration: none;
  z-index: 1;
}

.character-card-name-box {
  background-color: rgba(0, 0, 0, 0.65);
  width: 100%;
}

.character-card-text {
  height: 100%;
  margin: 0px 5px;
}

.card-link {
  margin: 0px 30px;
  width: 190px;
}