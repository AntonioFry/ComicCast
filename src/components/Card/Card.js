import React from 'react';
import './Card.css'

const Card = ({ name, id, image }) => {
  return (
    <div className="character-card" style={{backgroundImage: `url(${image})`}}>
      {/* <img className='character-card-img' src={image} alt='character'/> */}
      <div className="character-card-name-box">
        <h3 className="character-card-text">{name}</h3>
      </div>
    </div>
  )
}

export default Card;