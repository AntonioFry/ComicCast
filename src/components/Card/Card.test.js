import React from 'react';
import { Card } from './Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    wrapper = shallow(<Card />)
    props = {
      id: 1,
      name: 'man person',
      image: 'someimage.jpg'
    }
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})