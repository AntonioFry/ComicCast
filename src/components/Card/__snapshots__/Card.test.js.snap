// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
<Link
  to="/undefined"
>
  <div
    className="character-card"
    style={
      Object {
        "backgroundImage": "url(undefined)",
      }
    }
  >
    <div
      className="character-card-name-box"
    >
      <h3
        className="character-card-text"
      />
    </div>
  </div>
</Link>
`;
